docs/source/ProgrammingGuide/Initialization.md

```diff
@@ -43,66 +45,66 @@ Kokkos chooses the two spaces using the following list:
+--kokkos-num-threads=INT     | specify total number of threads to use for parallel regions on the host
+--kokkos-map-device-id-by=(random\|mpi_rank) | strategy to select device-id automatically from available devices: random or mpi_rank<sup>2</sup>
+--kokkos-tools-libs=STR      | specify which of the tools to use. Must either be full path to library or name of library if the path is present in the runtime library search path (e.g. LD_LIBRARY_PATH)
+--kokkos-tools-help          | query the (loaded) kokkos-tool for its command-line option support (which should then be passed via --kokkos-tools-args="...")
+--kokkos-tools-args=STR      | a single (quoted) string of options which will be whitespace delimited and passed to the loaded kokkos-tool as command-line arguments. E.g. `<EXE> --kokkos-tools-args="-c input.txt"` will pass `<EXE> -c input.txt` as argc/argv to tool
+
+When passing a boolean as a string, the acceptable values are:
+ - true, yes, 1
+ - false, no, 0
+
 
 ***
 <sup>1</sup> This is the preferred set of defaults when CUDA and OpenMP are enabled. If you use a thread-parallel host execution space, we prefer Kokkos' OpenMP back-end, as this ensures compatibility of Kokkos' threads with the application's direct use of OpenMP threads. Kokkos cannot promise that its Threads back-end will not conflict with the application's direct use of operating system threads.
+<sup>2</sup> The two device-id mapping strategies are:
+- random: choose a random device from available.
+- mpi_rank: choose device-id based on a round robin assignment of local MPI ranks. Works with OpenMPI, MVAPICH, SLURM, and derived implementations. Support for MPICH was added in Kokkos 4.0
 
-## 5.2 Initialization by struct
+## 5.2 Initialization by environment variable
 
-Instead of giving [`Kokkos::initialize()`](../API/core/initialize_finalize/initialize) command-line arguments, one may directly pass in initialization parameters using the following struct:
+Instead of using command-line arguments, one may use environment variables. The environment variables are identical to the arguments in Table 5.1 but they are upper case and the dash is replaced by an underscore. For example, if we want to set the number of threads to 3, we have
+  KOKKOS_NUM_THREADS=3
+
+## 5.3 Initialization by struct
+
+Instead of giving [`Kokkos::initialize()`](../API/core/initialize_finalize/initialize) command-line arguments, one may directly pass in initialization parameters using the `Kokkos::Initialization` struct.  If one wants to set a options using the struct, one can use the set functions `set_xxx` where `xxx` is the identical to the arguments in Table 5.1 where the dash has been replaced by an unsderscore. To check if a variable has been set, one can use the `has_xxx` functions. Finally, to get the value that was set, one can use the `get_xxx` functions.
```

Comment:
```suggestion
Instead of giving [`Kokkos::initialize()`](../API/core/initialize_finalize/initialize) command-line arguments, one may directly pass in initialization parameters using the `Kokkos::InitializationSettings` class.  If one wants to set a options using the struct, one can use the set functions `set_xxx` where `xxx` is the identical to the arguments in Table 5.1 where the dash has been replaced by an unsderscore. To check if a variable has been set, one can use the `has_xxx` functions. Finally, to get the value that was set, one can use the `get_xxx` functions.
```